ui/dashboards/src/components/Variables/VariableList.tsx

```diff
@@ -24,51 +24,197 @@ import {
   Table,
   Paper,
   TableHead,
+  Switch,
+  Typography,
+  IconButton,
 } from '@mui/material';
-import { useTemplateVariableDefinitions, useEditMode } from '../../context';
+import { VariableDefinition } from '@perses-dev/core';
+import { useImmer } from 'use-immer';
+import EyeIcon from 'mdi-material-ui/Eye';
+import PencilIcon from 'mdi-material-ui/Pencil';
+import TrashIcon from 'mdi-material-ui/TrashCan';
+
+import { useTemplateVariableDefinitions, useEditMode, useTemplateVariableActions } from '../../context';
 import { TemplateVariable } from './Variable';
+import { VariableEditForm } from './VariableEditorForm';
+
+function VariableEditor(props: {
```

Comment:
Cool sounds good 

@LukeTillman @eunicorn On a more general note, what do we consider long? Maybe we can just add the [max-lines eslint rule ](https://eslint.org/docs/latest/rules/max-lines) (seems to default to 300) if we have a general rule. Also, do we prefer one component per file? Personal preference is to avoid [too many files](https://twitter.com/dan_abramov/status/1220029098543788032?lang=en) (😆) but happy to adopt that as well if that's the general consensus.

Reply:
I personally prefer one component per file and name it the same as the component so it's easy to see/find what components there are, but I think it's ok that the file is long if the component is huge

Reply:
Yeah, I'm the same on preferring one component for file unless the second component is pretty small (e.g. a styled component, a really short helper component that's 20ish lines, etc). I just find it easier to read and jump around to other components in new tabs without losing my place in the original one if I want to go back (I also probably suck at navigating in the IDE tricks 😂).

My "this is getting pretty long I should split this up so others can read it" alarm tends to start warning me around 150-200 lines, but that's totally subjective. 🤷‍♂️